src/AWS.Deploy.Orchestration/OptionSettingHandler.cs

```diff
@@ -26,9 +26,47 @@ public OptionSettingHandler(IValidatorFactory validatorFactory)
         /// Thrown if one or more <see cref="Validators"/> determine
         /// <paramref name="value"/> is not valid.
         /// </exception>
-        public void SetOptionSettingValue(OptionSettingItem optionSettingItem, object value)
+        public void SetOptionSettingValue(Recommendation recommendation, OptionSettingItem optionSettingItem, object value)
         {
-            optionSettingItem.SetValue(this, value, _validatorFactory.BuildValidators(optionSettingItem));
+            optionSettingItem.SetValue(this, value, _validatorFactory.BuildValidators(optionSettingItem), recommendation);
+
+            // If the optionSettingItem came from the selected recommendation's deployment bundle,
+            // set the corresponding property on recommendation.DeploymentBundle
+            SetDeploymentBundleProperty(recommendation, optionSettingItem, value);
+        }
+
+        /// <summary>
+        /// Sets the corresponding value in <see cref="DeploymentBundle"/> when the
+        /// corresponding <see cref="OptionSettingItem"> was just set
+        /// </summary>
+        /// <param name="recommendation">Selected recommendation</param>
+        /// <param name="optionSettingItem">Option setting that was just set</param>
+        /// <param name="value">Value that was just set, assumed to be valid</param>
+        private void SetDeploymentBundleProperty(Recommendation recommendation, OptionSettingItem optionSettingItem, object value)
+        {
+            switch (optionSettingItem.Id)
+            {
+                case "DockerExecutionDirectory":
+                    recommendation.DeploymentBundle.DockerExecutionDirectory = value.ToString() ?? string.Empty;
+                    break;
+                case "DockerBuildArgs":
+                    recommendation.DeploymentBundle.DockerBuildArgs = value.ToString() ?? string.Empty;
+                    break;
+                case "ECRRepositoryName":
+                    recommendation.DeploymentBundle.ECRRepositoryName = value.ToString() ?? string.Empty;
+                    break;
+                case "DotnetBuildConfiguration":
+                    recommendation.DeploymentBundle.DotnetPublishBuildConfiguration = value.ToString() ?? string.Empty;
+                    break;
+                case "DotnetPublishArgs":
+                    recommendation.DeploymentBundle.DotnetPublishAdditionalBuildArguments = value.ToString() ?? string.Empty;
+                    break;
+                case "SelfContainedBuild":
+                    recommendation.DeploymentBundle.DotnetPublishSelfContainedBuild = (bool)value;
```

Comment:
Instead of casting can we do `System.Convert.ToBoolean(value)` to make the code more defensive.

Reply:
Done